File: frontend/src/types/interview.ts
export interface InterviewQuestion {
  id: string;
  question_text: string;
  category: string;
  expected_points: string[];
}

export interface InterviewSessionResponse {
  session_id: string;
  role_title: string;
  difficulty: string;
  questions: InterviewQuestion[];
}

export interface InterviewStartRequest {
  job_id?: string;
  role_title?: string;
  difficulty?: string;
}

export interface InterviewAnswerRequest {
  session_id: string;
  question_id: string;
  answer_text: string;
}

export interface AnswerEvaluationResponse {
  question_id: string;
  score: number;
  correctness: number;
  clarity: number;
  depth: number;
  feedback: string;
  missing_concepts: string[];
  grounded_suggestion: string;
}

File: backend/app/schemas/interview.py
"""Pydantic schemas for Adaptive Interview Simulator."""

import uuid
from typing import Any

from pydantic import BaseModel, Field


class InterviewStartRequest(BaseModel):
    job_id: uuid.UUID | None = None
    role_title: str = "Senior Distributed Systems Engineer"
    difficulty: str = "Senior"


class InterviewQuestion(BaseModel):
    id: str
    question_text: str
    category: str
    expected_points: list[str]


class InterviewSessionResponse(BaseModel):
    session_id: uuid.UUID
    role_title: str
    difficulty: str
    questions: list[InterviewQuestion]


class InterviewAnswerRequest(BaseModel):
    session_id: uuid.UUID
    question_id: str
    answer_text: str = Field(..., min_length=5)


class AnswerEvaluationResponse(BaseModel):
    question_id: str
    score: int
    correctness: int
    clarity: int
    depth: int
    feedback: str
    missing_concepts: list[str]
    grounded_suggestion: str
